search/search.go

```diff
@@ -36,9 +36,14 @@ func (ap ArrayPositions) Equals(other ArrayPositions) bool {
 }
 
 type Location struct {
+	// Pos is the position of the term within the field, starting at 1
 	Pos            uint64         `json:"pos"`
+	
+	// Start and End are the character offsets of the term in the field
 	Start          uint64         `json:"start"`
 	End            uint64         `json:"end"`
+	
+	// ArrayPositions contains the positions of the term within each array of subdocuments.
```

Comment:
I'd prefer we just say "...positions of the term within any elements."  The reason is that it can be for simple values as well, not just subdocuments, and the term sub-documents seems to be used  in different ways by different people, so I try to avoid it.